core/src/main/scala/cats/syntax/monad.scala

```diff
@@ -20,3 +24,28 @@ final class MonadIdOps[A](private val a: A) extends AnyVal {
    */
   def iterateUntilM[F[_]](f: A => F[A])(p: A => Boolean)(implicit M: Monad[F]): F[A] = M.iterateUntilM(a)(f)(p)
 }
+
+final class MonadBooleanOps[F[_]](private val fa: F[Boolean]) extends AnyVal {
+
+  /**
+   * Behaves like `&&` but inside the `F` context.
+   *
+   * Wont evaluate `other` unless this evaluates to `true`
+   */
+  def andM(other: => F[Boolean])(implicit F: Monad[F]): F[Boolean] =
+    F.flatMap(fa) {
+      case true  => other
+      case false => F.pure(false)
+    }
+
+  /**
+   * Behaves like `||` but inside the `F` context.
+   *
+   * Wont evaluate `other` unless this evaluates to `false`
```

Comment:
This is wrong. Also underscores the need for tests.

Reply:
Care to elaborate why that is wrong?

Reply:
Or returns true if either are true. This returns false if the first is true.

Reply:
ups... that happens for copypasting code 😅
I am sorry for that.

Yeah I agree tests are good, but then again how should I test them? The truth tables?

Reply:
I would say that for all pairs of inputs `Id[Boolean]` ops should behave like the normal ops on Boolean, which yes is equivalent to testing the truth table.